### PR TITLE
Support Apple silicon

### LINKED ISSUE #828: Provide release build for arm64

### PR DESCRIPTION
**Pull request by szepeviktor**
Fixes #828 

@epage Is this the right direction? I've never seen a Rust environment.

### CODE REVIEW

File: .github/workflows/post-release.yml [hunk 1]
```diff
@@ -89,6 +93,13 @@ jobs:
       with:
         toolchain: ${{ matrix.rust }}
         targets: ${{ matrix.target }}
+    - name: Set up Apple silicon support
+      if: matrix.target == 'aarch64-apple-darwin'
+      shell: bash
+      run: |
+        rustup target add aarch64-apple-darwin
```
Comment (epage) [inline .github/workflows/post-release.yml:100]: Won't the `rust-toolchain` step above take care of this?

Comment (szepeviktor) [reply .github/workflows/post-release.yml:100]: Sorry, I do not know.
I have zero knowledge of Rust environments.

**Comment by szepeviktor** [comment]
This PR is really an inspiration for you.

**Comment by szepeviktor** [comment]
I think the amd64 binary was built: https://github.com/szepeviktor/typos/actions/runs/7493064842/job/20397945540#step:6:1
I leave it up to you!

**Pull request closed without merging**